tests/python/relay/aot/aot_test_utils.py

```diff
@@ -731,130 +733,129 @@ def run_and_check(
     This method uses the original test data and compiled runtime.Modules
     to run in the test runner to verify the results.
     """
+    tmpdir = tempfile.TemporaryDirectory() if test_dir is None else contextlib.surpress()
```

Comment:
Should not this be https://docs.python.org/3/library/contextlib.html#contextlib.suppress ?

Also why do we need to suppress execeptions here ?

I think a slightly better approach would be to make the body of "with" to be wrapped in a function that accepts base_path, where the dir could be provided within a "with" if test_dir is None. WDYT ?

Reply:
Yeah this was an attempt to make the `with` statement conditional, but without limiting us to Python 3.7 (in which case `nullcontext` could have been used), see https://stackoverflow.com/a/45187287. I agree it looks a bit strange, I was trying to avoid wrapping the body in another function. In hindsight it probably would be more readable to do that though.